erpnext/crm/doctype/contract/contract.py

```diff
@@ -284,6 +289,52 @@ def get_contract_list(doctype, txt, filters, limit_start, limit_page_length=20,
 			""".format(limit_start, limit_page_length), [contracts], as_dict=1)
 
 
+def send_contract(contract, method):
+	if method == "on_submit":
+		recipients = [party_user.user for party_user in contract.party_users]
+
+		if recipients:
+			# TODO: Change to generic settings for pushing to ERPN
+			settings = frappe.get_single("JH Audio Settings")
+			subject = settings.contract_email_subject
+			message = settings.contract_email_message
+
+			if not (subject and message):
+				frappe.throw(_("Please enter a subject and message in JH Audio Settings to send contracts"))
+			else:
+				contract_data = {
+					"recipients": recipients,
+					"subject": subject,
+					"content": message,
+					"doctype": "Contract",
+					"name": contract.name
+				}
+
+				# Add sales partner as BCC, by pulling their email either through a
+				# linked User record, or by matching a System User with the same name
+				if contract.sales_partner:
+					sales_partner_email = frappe.db.get_value("Sales Partner", contract.sales_partner, "user")
+
+					if not sales_partner_email:
+						sales_partner_email = frappe.db.get_value("User", filters={"full_name": contract.sales_partner, "user_type": "System User"})
+
+					if sales_partner_email:
+						contract_data.update({"bcc": [sales_partner_email]})
+
+				_send_contract(contract.name, contract_data)
+
+
+def _send_contract(contract_name, contract_data):
+	print_format = frappe.db.get_value("Print Format", filters={"doc_type": "Contract"})
+	attachments = [frappe.attach_print("Contract", contract_name, print_format=print_format)]
```

Comment:
Just confirming, does this method attach it in the body or sends it dynamically? Need you to check. we need to actively avoid sending out emails with the PDF in the body.

Reply:
Yeah, this attaches it to the body. I'll look at a way to pull just the file name.